src/lib.rs

```diff
@@ -248,3 +251,21 @@ pub use result::{IonError, IonResult};
 pub mod external {
     pub use bigdecimal;
 }
+
+/// Whether or not the text spacing is generous/human-friendly or something more compact.
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+#[non_exhaustive]
+pub enum TextKind {
+    Compact,
+    Lines,
+    Pretty,
+}
+
+/// Supported Ion encodings.
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+#[non_exhaustive]
+pub enum Format {
+    Text(TextKind),
+    Binary,
+    // TODO: Json(TextKind)
+}
```

Comment:
Are these useful for anything if you haven't opted into the `experimental-writer` feature? If not, we should move them to a `pub(crate)` module and conditionally re-export them only if the feature is enabled.

Reply:
They're used in the `Element::write_as` API:
```rust 
let mut buffer = Vec::new();
Element::write_as(Format::Text(TextKind::Pretty), &mut buffer)?;
// ...do stuff with `buffer`
```